python/tests/io/test_pyarrow.py

```diff
@@ -18,73 +18,19 @@
 
 import os
 import tempfile
-from typing import Any, List, Optional
```

Comment:
Oeeeh, thanks for splitting this. It was getting too big indeed. A lot of rebasing for me ahead